source/javascripts/components/ProductTour/ProductTour.tsx

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import React, { useEffect } from "react";
+import { useCallback, useState } from "react";
+import { Highlighter, useHighlighter } from "./Highlighter";
+import { ProductTooltip } from "./ProductTooltip";
+import { ProductTourProps, Tips } from "./types";
+import { useTrackingFunction } from "./useTrackingFunction";
+import { tips } from "./tips";
+import { useWaitForElements } from "./useWaitForElement";
+
+export const ProductTourContent = ({ menuIds, currentUser }: ProductTourProps): JSX.Element | null => {
+	const [isOpen, setIsOpen] = useState(true);
+	const [validTips, setValidTips] = useState<Tips[] | null>(null);
+	const [selectedId, setSelectedId] = useState("");
+
+	const onFound = useCallback(() => {
+		const filtered = tips.filter(tip => menuIds.includes(tip.id));
+		setValidTips(filtered);
+		setSelectedId(filtered[0].id);
+	}, []);
+
+	// NOTE: Angular passes the menu items ids it will render,
+	// but we stil need to wait for the templates to render them before we can move on.
+	useWaitForElements(menuIds, onFound);
+
+	const { rect, clipPath, highlighted } = useHighlighter(selectedId);
+
+	const onDisplayTooltip = useTrackingFunction(() => ({
+		event: "tooltip_displayed",
+		payload: {
+			user_slug: currentUser.slug,
+			user_id: currentUser.dataId,
+			location: "workflow_editor",
+			name: selectedId
+		}
+	}));
+
+	const onTrackClose = useTrackingFunction(() => ({
+		event: "tooltip_closed",
+		payload: {
+			user_slug: currentUser.slug,
+			user_id: currentUser.dataId,
+			location: "workflow_editor",
+			name: selectedId
+		}
+	}));
+
+	const onButtonClick = useTrackingFunction((buttonName: string) => ({
+		event: "tooltip_clicked",
+		payload: {
+			user_slug: currentUser.slug,
+			user_id: currentUser.dataId,
+			location: "workflow_editor",
+			name: selectedId,
+			button: buttonName
+		}
+	}));
+
+	const onClose = (): void => {
+		setIsOpen(false);
+		onTrackClose();
+	};
+
+	useEffect(() => {
+		if (isOpen && !!selectedId && highlighted) {
+			onDisplayTooltip();
+		}
+	}, [isOpen, highlighted, onDisplayTooltip, selectedId]);
+
+	const onChange = useCallback(id => {
```

Comment:
It is currently, because of this: 
```
useEffect(() => {
		if (selectedId !== undefined) {
			onChange(selectedId);
		}
	}, [selectedId, onChange]);
```
But I don't like it this way so I'll take another look

Reply:
I've refactored it and lifted up the state to `ProductTour` to not have the same state - who is selected - be replicated in  the 2 places. This solves this super bad syncing useEffect too.